src/pytest_html/resources/main.js

```diff
@@ -244,3 +247,83 @@ function filterTable(elem) { // eslint-disable-line no-unused-vars
     const notFoundMessage = document.getElementById('not-found-message');
     notFoundMessage.hidden = !allRowsHidden;
 }
+
+var PieChart = function () {
```

Comment:
@xiaosanye886 
did you write this code or is it taken from some library? (which lib?)

Reply:
no, i'm not good at JS, so about the canvas to draw  a pie chart， I learn from https://blog.csdn.net/qq_38944959/article/details/88321304

Reply:
@jkowalleck @xiaosanye886 this may be impacted by the impending re-write and it could make sense to hold off on this until it's done.

I'm busy with work lately so I won't have time to dedicate to it in the next couple of months,  but perhaps @BeyondEvil can shed some more light on this when he gets the time ( since he's busy himself lately ).